pyiron_atomistics/atomistics/structure/atoms.py

```diff
@@ -1337,37 +1316,38 @@ def get_neighbors_by_distance(
         return self.get_neighbors(
             cutoff_radius=cutoff_radius,
             num_neighbors=num_neighbors,
-            t_vec=t_vec,
             tolerance=tolerance,
             id_list=id_list,
             width_buffer=width_buffer,
             allow_ragged=allow_ragged,
+            mode=mode,
             norm_order=norm_order,
         )
 
+    @deprecate(allow_ragged="use mode instead.")
     def get_neighbors(
         self,
         num_neighbors=12,
-        t_vec=True,
         tolerance=2,
         id_list=None,
         cutoff_radius=np.inf,
         width_buffer=1.2,
-        allow_ragged=False,
+        allow_ragged=None,
+        mode='filled',
         norm_order=2,
     ):
         """
 
         Args:
             num_neighbors (int): number of neighbors
-            t_vec (bool): True: compute distance vectors
-                        (pbc are automatically taken into account)
             tolerance (int): tolerance (round decimal points) used for computing neighbor shells
             id_list (list): list of atoms the neighbors are to be looked for
             cutoff_radius (float): Upper bound of the distance to which the search must be done
             width_buffer (float): width of the layer to be added to account for pbc.
-            allow_ragged (bool): Whether to allow ragged list of arrays or rectangular
-                numpy.ndarray filled with np.inf for values outside cutoff_radius
+            allow_ragged (bool): (Deprecated; use mode) Whether to allow ragged list of arrays or
+                rectangular numpy.ndarray filled with np.inf for values outside cutoff_radius
+            mode (str): Representation of per-atom quantities (distances etc.). Choose from
```

Comment:
The behaviour of the different modes should also be documented. Here at the variable definition is fine, but since it's longer you could also have something like `"Neighbor data is formatted to be returned in one of three ways based on the mode argument. The first and default format is 'filled'..."`